net/HttpRequest.hpp

```diff
@@ -1692,14 +1695,18 @@ class Session final : public ProtocolHandlerInterface
         net::asyncConnect(_host, _port, isSecure(), shared_from_this(), pushConnectCompleteToPoll);
     }
 
-    void checkTimeout(std::chrono::steady_clock::time_point now) override
+    bool checkTimeout(std::chrono::steady_clock::time_point now) override
```

Comment:
Why is this needed at all? This function changes the state of the and fires up the necessary callbacks (see `_response->timeout();` below). We can easily check the state and act accordingly. And anyway, the loop above (line 1435) will be broken.

Reply:
In general I prefer a clear response from a method, here true for timeout in checkTimeout() -> also changed API doc.

Usually this method is called in Socket::handlePoll() where I prefer to bail out quick.
In general I prefer `fail fast` and `bail out early` to help reducing complexity. However, I understand that there might be more side effects and `fail fast` often stands against `keep going`. Hence this review is important, thx!

The particular case of http::Session::syncRequestImpl()  .. se below .. probably needs to change indeed, great catch.